travis: backport 10.3 changes

grooverdan opened the pull request:
Also recommend enabling Travis auto-cancellation to prevent builds queuing up on the same branch if it isn't already. https://blog.travis-ci.com/2017-09-21-default-auto-cancellation.

Lets wait for this to finish. There's a substantial backlog of osx builds for the last day.

svoj commented on debian/rules:
```diff
@@ -32,7 +32,7 @@ CXX := $(DEB_HOST_GNU_TYPE)-g++
 ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
     NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
-    MAKEFLAGS += -j $(NUMJOBS)
+    MAKEFLAGS += -j$(NUMJOBS)
```
This was recently done by @ottok: 325c9ce9b39a7695bf217cd92fc36034b2e8207b

grooverdan replied:
apparently this was just re-conforming to the debian policy docs with no material impacts.

svoj commented:
Taking into account recent activity around travis, I wonder who is to review this? @ottok @cvicentiu ?

grooverdan commented:
Not quite following the milestones being a 10.2 target.

@cvicentiu commented ... https://github.com/MariaDB/server/pull/499#issuecomment-345993346

FYI - also testing https://travis-ci.org/grooverdan/mariadb-server/builds/309873867 based on few Travis bug comments.

svoj commented:
Sorry, my bad.

codership-jenkins commented:
Can one of the admins verify this patch?

grooverdan commented:
Backported the 10.3 changes here. Added percona which is still a test in 10.2. Predictable failures on  rocksdb.mariadb_plugin and sys_vars.innodb_buffer_pool_load_at_startup_basic however all the other 10.3 fixes are included.

I added a few trivial patches at the end too.

grooverdan commented:
https://github.com/MariaDB/server/pull/559#issuecomment-402417109 "Right now our biggest priority is to backport these Travis-CI improvements". Obvious not, since this still hasn't been merged. It doesn't resolve all 10.2 issues however it fix a lot of them.

svoj commented:
@shinnok could you have a look at this if there's still anything to backport?
